addons/tls13-comparison-v1/bootstrap.js

```diff
@@ -0,0 +1,90 @@
+/* -*- indent-tabs-mode: nil; js-indent-level: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+let {classes: Cc, interfaces: Ci, utils: Cu} = Components;
+
+Cu.import("resource://gre/modules/Preferences.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/TelemetryController.jsm");
+Cu.import("resource://gre/modules/ClientID.jsm");
+
+Cu.importGlobalProperties(["crypto", "TextEncoder"]);
+const DEBUG = false;
+const ENABLE_PROB = .5;
+let gStarted = false;
+const kVERSION_MAX_PREF = "security.tls.version.max";
+
+function debug(msg) {
+    if (DEBUG) {
+        console.log(`TLS 1.3 Test: ${msg}`);
+    }
+}
+
+function disable() {
+  Experiments.instance().disableExperiment("FROM_API");
+}
+
+async function getRandomnessSeed() {
+  return await ClientID.getClientID();
+}
+
+function toHex(arr) {
+  let hexCodes = [];
+  for (let i = 0; i < arr.length; ++i) {
+    hexCodes.push(arr[i].toString(16).padStart(2, "0"));
+  }
+
+  return hexCodes.join("");
+}
+async function generateVariate(seed, label) {
+  const hasher = crypto.subtle;
+  const hash = await hasher.digest("SHA-256", new TextEncoder("utf-8").encode(seed + label));
+  let view = new DataView(hash);
+  return view.getUint32(0) / 0xffffffff;
+}
+
+async function startup(addon) {
```

Comment:
Nit: the first optional param to `startup` is usually called `data` (the second optional one being `reason`).